### ISSUE TITLE
Style/Documentation: False positive on module documentation comments.

### ISSUE THREAD

**Issue by woodruffw** [bbatsov/rubocop#3471]
I came across this behavior while adding RuboCop to [ruby-macho](https://github.com/Homebrew/ruby-macho).

## Expected behavior

RuboCop not complaining about a missing top-level module documentation comment when one has been already provided in another file.

## Actual behavior

RuboCop complains about module `MachO` missing a top-level comment in file *lib/macho/open.rb*, even though one was already provided in *lib/macho.rb*.

## Steps to reproduce the problem

```bash
git clone https://github.com/Homebrew/ruby-macho.git
cd ruby-macho
rubocop -DES lib/
```

Output:
```
Inspecting 12 files
........C...

Offenses:

lib/macho/open.rb:1:1: C: Style/Documentation: Missing top-level module documentation comment. 
module MachO
^^^^^^

12 files inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-linux-gnu)
```

**Comment by maxjacobson** [comment]
RuboCop doesn't consider files in the context of a broader codebase, so it currently doesn't have a way of knowing that a module was documented in another file.

This cop will, however, skip a module if it believes it to be purely a namespace, which it defines like this [(via the cop's documentation)](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/documentation.rb):

```
      # This cop checks for missing top-level documentation of
      # classes and modules. Classes with no body are exempt from the
      # check and so are namespace modules - modules that have nothing in
      # their bodies except classes, other modules, or constant definitions.
      #
      # The documentation requirement is annulled if the class or module has
      # a "#:nodoc:" comment next to it. Likewise, "#:nodoc: all" does the
      # same for all its children.
```

So I think you have 2 workarounds -- either refactor your code to make it meet the cop's definition of a namespace module, or add a nodoc comment. Does that work for you?

**Comment by woodruffw** [comment]
> So I think you have 2 workarounds -- either refactor your code to make it meet the cop's definition of a namespace module, or add a nodoc comment. Does that work for you?

It does, thank you. I'll close this.

**Comment by ifreecarve** [comment]
Just to be explicit here: if this cop is triggered, it may indicate that you have a stray method or constant outside a class definition.

```ruby

module MyModule

  # Doc for the class.  
  # The module is documented in the root-level file where it's first referenced,
  #   so it's not documented here.
  class MyClass

    # my method doc
    def my_method
        nil
    end

  #   now imagine many more lines of definitions here 
  #
  #         end
  #       end
  #     end
  #   end
  # end
  end

  # this caused my rubocop error -- this should have been in the class definition
  # were it not for a bad copy/paste during refactoring
  def another_method
    "whoops"
  end
end
```
